chore!: Flatten upload options object

CallumGrindle opened the pull request:
### What kind of change does this PR introduce (bug fix, feature, docs update, ...)?
Flattens `uploadOptions` parameters to just `ipfsGateway`

Before:
```javascript
await lspFactory.LSP7DigitalAsset.deploy({...}, {
  uploadOptions: {
    ipfsClientOptions: {
      host: 'ipfs.infura.io',
      port: 5001,
      protocol: 'https',
    }
  }
});
```

After: 
```javascript
await lspFactory.LSP7DigitalAsset.deploy({...}, {
  ipfsGateway: {
    host: 'ipfs.infura.io',
    port: 5001,
    protocol: 'https',
  }
});
```

ipfsGateway can also be passed as a string:
```javascript
await lspFactory.LSP7DigitalAsset.deploy({...}, {
  ipfsGateway: "https://ipfs.infura.io:5001"
});
```

```javascript
await lspFactory.LSP7DigitalAsset.deploy({...}, {
  ipfsGateway: "https://ipfs.infura.io"
});
```

If no port is provided, port will be set to 5001

See example at https://docs.lukso.tech/tools/lsp-factoryjs/deployment/universal-profile#ipfs-upload-options

The ipfsClientOptions object is passed directly to the IPFS client. The available options can be seen here https://github.com/ipfs/js-ipfs/tree/master/packages/ipfs-http-client#usage

Hugoo reviewed (approved):
cool - do you also need to update the docs to explain this new param?

frozeman commented:
Can you please describe in the PR description what you changed how? i want o prevent more rounds of back and forth.

As it looks like you simple changed the name from `uploadOptions` to `ipfsClientOptions`, but thats not what we agreed on.

Can you please show what the `ipfsClientOptions` look like?

It should be simply `ipfsGateway` with a string as value.
Ideally it also accepts an object as value with `url, port` and potentially others.

This way its flexible and extensible without breaking the API.

frozeman commented:
Lets please rename it to `ipfsGateway` to be consistent with erc725.js and allow both the object and a URL string to be passed in.

You can derive the port from the URL or present, otherwise use the standard port `5001`

frozeman reviewed (changes requested):
see comment above

frozeman commented:
Ps. i dont understand the timeouts in the tests, are we not mocking transaction calls?

CallumGrindle commented:
@frozeman I have updated PR with changes. 

We use hardhat for transactions. I had problems with the tests due to IPFS issues so increased timeouts. They are unnecessary now so I have reverted

CJ42 commented on docs/deployment/digital-asset.md:
```diff
@@ -379,23 +379,37 @@ await lspFactory.LSP8IdentifiableDigitalAsset.deploy({...}, {
 ### IPFS Upload Options
 
-You can specify how you want your profile metadata to be uploaded by passing the `ipfsClientOptions` inside the `options` object. Here you can set the IPFS gateway where you want the metadata to be uploaded.
+You can specify how you want your profile metadata to be uploaded by passing the `ipfsGateway` inside the `options` object. Here you can set the IPFS gateway where you want the metadata to be uploaded.
```
**Suggested change:**

You can specify the IPFS network where you want to upload your profile metadata using an `ipfsGateway` object inside the `options` parameter.

CJ42 commented on docs/deployment/digital-asset.md:
```diff
@@ -379,23 +379,37 @@ await lspFactory.LSP8IdentifiableDigitalAsset.deploy({...}, {
 
 :::note
-The options object takes an `ipfsClientOptions` object as defined by the [IPFS-HTTP Client](https://github.com/ipfs/js-ipfs/tree/master/packages/ipfs-http-client#createoptions) library which is used internally to interact with the specified IPFS node.
+The procedure takes a URL string or an object as defined by the [IPFS-HTTP Client](https://github.com/ipfs/js-ipfs/tree/master/packages/ipfs-http-client#createoptions) library which is used internally to interact with the specified IPFS node.
```
Suggested change:

`procedure` -> `ipfsGateway` option

CJ42 commented on docs/getting-started.md:
```diff
@@ -110,6 +110,44 @@ You can now continue using your UP address within the dApp:
 
+## Options
+
+When instantiating LSPFactory options can be passed to specify parameters such as `chainId` and `ipfsGateway`.
```
Suggested change:

_"The following parameters are available when creating an instance of the LSPFactory"_

CJ42 commented on docs/getting-started.md:
````diff
@@ -110,6 +110,44 @@ You can now continue using your UP address within the dApp:
+});
+```
+
+#### Deploy Key
````
Maybe a small table could be useful as a summary of the available options?

Like here: https://github.com/cgewecke/eth-gas-reporter#options

CJ42 reviewed (commented):
added suggestion comments
